airbyte_rock/rockcraft.yaml

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: airbyte
+summary: Airbyte rock
+description: Airbyte OCI image for the Airbyte charm
+version: "1.0"
+base: ubuntu@22.04
+build-base: ubuntu@22.04
+license: Apache-2.0
+platforms:
+  amd64:
+
+environment:
+  JAVA_HOME: /usr/lib/jvm/java-21-openjdk-amd64
+  CDK_PYTHON: /usr/bin/python3.10
+  CDK_ENTRYPOINT: /usr/lib/python3.10/dist-packages/airbyte_cdk/connector_builder/main.py
+  CDK_VERSION: "5.12.0"
+
+
+parts:
+  install-dependencies:
+    plugin: nil
+    stage-packages:
+      - apt-transport-https
+      - ca-certificates
+      - curl 
+      - gnupg
+      - python3.10-venv  
+    override-build: |
+      # Install kubectl
+      echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.31/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list
+      chmod 644 /etc/apt/sources.list.d/kubernetes.list   # helps tools such as command-not-found to work correctly
+      curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.31/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
+      chmod 644 /etc/apt/keyrings/kubernetes-apt-keyring.gpg # allow unprivileged APT programs to read this keyring
+      apt-get update
+      apt-get install -y kubectl
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/local/bin
+      cp -r $(which kubectl) ${CRAFT_PART_INSTALL}/kubectl
+
+      pip install --upgrade setuptools pip airbyte-cdk==5.12.0 \
+          --target=/${CRAFT_PART_INSTALL}/dist
+
+    organize:
+      kubectl: usr/local/bin/kubectl
+      dist: usr/local/lib/python3.10/dist-packages
+    stage:
+      - usr/local/bin/kubectl
+      - usr/local/lib/python3.10/dist-packages
+
+  pull-airbyte-repo:
+    after: [install-dependencies]
+    plugin: dump
+    source: https://github.com/airbytehq/airbyte-platform.git # yamllint disable-line
+    source-type: git
+    source-tag: v0.63.8
+    override-build: |
+      cp -r . ${CRAFT_PART_INSTALL}/airbyte-platform
```

Comment:
You don't need to copy the files manually since this is done already by the dump plugin. You can use `organize` to change the destination folder if needed.

Reply:
Trying to use `organize` did not seem to yield the expected result, reverted to this approach for now.

Reply:
@kelkawi-a  Apologies! It seems that the dump plugin copies just the contents but not the parent directory itself, hence you need that explicit copy.